### ISSUE TITLE
[Model Serving]: Create metrics wrappers and metrics provider for model serving

### ISSUE THREAD

**Issue by lucferbux** [opendatahub-io/odh-dashboard#825]
### Feature description

Create two wrapper components to fetch data for the respective context and mount a metric provider:
* Global View Serving Wrapper
* DSG Serving Wrapper

Create a metrics provider to get the queries for the metrics visualization.

**Comment by andrewballantyne** [comment]
### Frontend Architecture Structure

UI discussion meeting came up with this infrastructure for the metrics serving effort. These speak React components with a basic understanding of the way the current Model Serving setup is -- more will be explained below.

![image](https://user-images.githubusercontent.com/8126518/205697314-e8867510-9770-4275-990c-cda56f70158f.png)

#### High level style notes

- Two tickets to manage:
    - no1 is the "white background" blocks, and is this ticket
    - no2 is the "blue background" blocks, and is #656 
- The "red coloured text" share the same type
- Shapes are basically type of files:
    - Rectangles (90 degree corners) are straight basic React Components
    - Rectangles (curved corners) are React Context Providers
    - Circles are Utilities

#### Basically we have two flows

1. Inside the DS Projects (DSG) UI flows
2. The global Model Serving page

The difference mainly being how they fetch and store data. We have a context for each flow. The DSG context has all things dsg, where the model serving has all things model serving and expands outside of the one namespace model to show all serving in one spot. Because of this design, we need two wrappers -- to interact and normalize the context references.

`Metrics Provider` will be the point of normalization -- we should reference no other context state under this one. It should just create a storage for metrics data to be stored.

`Graphs A..D` just references whatever graphs we want to provide. I believe at this time there is only 4.

**Comment by andrewballantyne** [comment]
@DaoDaoNoCode I believe you have done this work -- can you assign this and associate a PR?

**Comment by lucferbux** [comment]
Yup, that's right, this effort is done by @DaoDaoNoCode in his PR.
